packages/big-design/src/components/Select/Select.tsx

```diff
@@ -1,715 +1,427 @@
+  // for single selects
+  const selectedOption = options.find(option => option.value === value);
+  // for multi selects
+  const selectedOptions =
+    multi && Array.isArray(value)
+      ? options.filter(option => value.find(val => val === option.value) !== undefined)
+      : [];
+
+  function renderLabel(getLabelProps: UseSelectPropGetters<Option<T>>['getLabelProps']) {
     return typeof label === 'string' ? (
-      <Label htmlFor={inputId} id={labelId} renderOptional={!required}>
+      <Label {...getLabelProps()} renderOptional={!required}>
         {label}
       </Label>
     ) : null;
   }
 
-  private renderInput() {
-    const {
-      name,
-      placeholder,
-      error,
-      filterable = true,
-      required,
-      disabled,
-      onItemChange,
-      options,
-      value,
-    } = this.props;
-    const { highlightedItem, inputText, isOpen } = this.state;
-    const ariaActiveDescendant = highlightedItem ? { 'aria-activedescendant': highlightedItem.id } : {};
-    const ariaControls = isOpen ? { 'aria-controls': this.getSelectId() } : {};
-    const id = this.getInputId();
-
-    const chips = this.renderChips();
-
-    const onChipDelete = (chip: string) => () => {
-      const filteredValues = Array.isArray(value)
-        ? value.filter(val => {
-            const foundOption = options.find(option => option.value === val);
-
-            return foundOption && foundOption.content !== chip;
-          })
-        : [];
-
-      onItemChange(filteredValues, this.getSelectedOptions(filteredValues));
-      this.focusInput();
-    };
-
+  function renderInput(
+    getInputProps: PropGetters<Option<T>>['getInputProps'],
+    getToggleButtonProps: UseSelectPropGetters<Option<T>>['getToggleButtonProps'],
+    inputValue: DownshiftState<Option<T>>['inputValue'],
+    openMenu: Actions<Option<T>>['openMenu'],
+  ) {
     return (
       <Reference>
         {({ ref }) => (
           <StyledInputContainer ref={ref}>
             <Input
-              aria-autocomplete="list"
-              autoComplete="off"
-              chips={chips}
-              disabled={disabled}
-              error={error}
-              iconRight={this.renderDropdownIcon()}
-              id={id}
-              name={name || id}
-              onChange={this.handleOnInputChange}
-              onChipDelete={chips && onChipDelete}
-              onClick={this.handleOnInputSelected}
-              onFocus={this.handleOnInputSelected}
-              onKeyDown={this.handleOnInputKeyDown}
-              placeholder={placeholder}
+              {...rest}
+              {...getInputProps({
+                disabled,
+                onClick: () => {
+                  openMenu();
+                },
+                onFocus: () => {
+                  openMenu();
+                },
+                onKeyDown: (event: any) => {
+                  if (multi && event.key === 'Backspace' && !inputValue) {
```

Comment:
Should we be checking for `inputValue.length`? Could there be a corner case with `0`?

Reply:
Hm, I fail to see why it needs to change, tbh. `inputValue` is a string, and will only return falsey if it's `""` right?